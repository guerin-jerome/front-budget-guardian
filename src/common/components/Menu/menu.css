.navbar {
  background-color: #ffffff;
  border-bottom: 1px solid #eeeeee;
  display: flex;
  justify-content: center;
}

.navbar-nav {
  display: flex;
  flex-direction: row;
  list-style: none;
  margin: 0;
  padding: 0;
}

.nav-item {
  align-items: center;
  display: flex;
  position: relative;
}

.nav-link {
  width: 200px;
  text-align: center;
  color: #666666;
  display: block;
  font-size: 14px;
  font-family: Arial, sans-serif;
  padding: 10px;
  position: relative;
  text-decoration: none;
  transition: all 0.3s ease-in-out;
}

.nav-link:after {
  background-color: #0077c2;
  content: "";
  height: 2.5px;
  left: 0;
  position: absolute;
  bottom: -2.5px;
  transform: scaleX(0);
  transition: transform 0.3s ease-in-out;
  width: 100%;
}

.nav-link.active {
  color: #0077c2;
}

.nav-link:hover:after,
.nav-link.active:after {
  transform: scaleX(1);
}

@media only screen and (max-width: 750px) {
  .nav-link {
    width: 33vw;
  }
}
