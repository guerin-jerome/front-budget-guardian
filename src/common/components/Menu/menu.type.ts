import { MouseEventHandler } from "react";

export type MenuItem = {
  text: string;
  href: string;
  isActive: boolean;
  onClick?: MouseEventHandler<HTMLAnchorElement>;
};

export type MenuProps = {
  items: MenuItem[];
};
