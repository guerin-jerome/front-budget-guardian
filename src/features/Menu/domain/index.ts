import { buildItemText } from "./buildItemText";

export { buildItemText };
