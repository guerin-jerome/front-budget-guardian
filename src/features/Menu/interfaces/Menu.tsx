import { Outlet } from "react-router-dom";
import { useMenu } from "../hooks/useMenu";
import classNames from "classnames";
import "./style.css";

export const Menu = () => {
  const { menuItems, onClickItem } = useMenu();

  return (
    <>
      <nav className="navbar">
        <ul className="navbar-nav">
          {menuItems.map(({ text, href, isActive }) => (
            <li className="nav-item" key={text}>
              <a
                onClick={(event) => onClickItem(event, href)}
                className={classNames("nav-link", { active: isActive })}
                href={href}
              >
                {text.toUpperCase()}
              </a>
            </li>
          ))}
        </ul>
      </nav>
      <Outlet />
    </>
  );
};
