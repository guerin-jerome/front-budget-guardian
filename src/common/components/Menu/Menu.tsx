import { MenuProps } from "./menu.type";
import "./menu.css";
import classNames from "classnames";

export const Menu = ({ items }: MenuProps) => {
  return (
    <nav className="navbar">
      <ul className="navbar-nav">
        {items.map(({ text, href, isActive, onClick }) => (
          <li className="nav-item" key={text}>
            <a
              onClick={onClick}
              className={classNames("nav-link", { active: isActive })}
              href={href}
            >
              {text.toUpperCase()}
            </a>
          </li>
        ))}
      </ul>
    </nav>
  );
};
