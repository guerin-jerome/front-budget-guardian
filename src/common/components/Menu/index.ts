import { Menu } from "./Menu";
import { MenuItem } from "./menu.type";

export { Menu };
export type { MenuItem };
