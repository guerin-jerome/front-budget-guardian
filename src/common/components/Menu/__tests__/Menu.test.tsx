import { render, screen, waitFor } from "@testing-library/react";
import { describe, it, expect, vi } from "vitest";
import { Menu } from "../Menu";
import userEvent from "@testing-library/user-event";
import { MouseEvent } from "react";

describe("<Menu />", () => {
  const navigate = vi.fn((event: MouseEvent<HTMLAnchorElement>) => {
    event.preventDefault();
  });

  const initialItems = [
    { text: "Test", href: "/test", isActive: false, onClick: navigate },
  ];

  it("matchs snapshot", () => {
    const { container } = render(<Menu items={initialItems} />);
    expect(container).toMatchSnapshot();
  });

  it("renders well and should navigate", async () => {
    render(<Menu items={initialItems} />);

    const navLink = screen.getByRole("link", { name: /test/i });
    expect(navLink).toBeDefined();

    await userEvent.click(navLink);

    expect(navigate).toHaveBeenCalled();
  });
});
