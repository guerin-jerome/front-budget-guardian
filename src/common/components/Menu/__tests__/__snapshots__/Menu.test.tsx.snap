// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Menu /> > matchs snapshot 1`] = `
<div>
  <nav
    class="navbar"
  >
    <ul
      class="navbar-nav"
    >
      <li
        class="nav-item"
      >
        <a
          class="nav-link"
          href="/test"
        >
          TEST
        </a>
      </li>
    </ul>
  </nav>
</div>
`;
